doc/nrf/ug_partition_manager.rst

```diff
@@ -279,3 +279,72 @@ CMake usage
+In these use cases, the images to be upgraded must be linked to the same address as that which is deployed.
+This is to avoid accidental overrides and incorrect configurations.
+
+Predefined configuration still allows the user to configure the **dynamic partition**.
+The dynamic partition consists of all memory adjacent to the "app" partition which is not occupied by a predefined partition.
+When Partition Manager is executed, it only operates on the dynamic partition, assuming that all other memory is reserved.
+Within the dynamic partition, it is allowed to define new partitions.
+The dynamic partition is re-sized by adding or removing partitions to the predefined configuration.
+
+.. _ug_pm_predefined_providing:
+
+Providing predefined configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The predefined configuration is provided through a YAML formatted configuration file.
+This file is similar to the regular :file:`pm.yml` configuration files, except
+that it also defines the start address for all partitions.
+The given configuration for a build can be found in :file:`${BUILD_DIR}/zephyr/include/generated/prj.partitions.yml`.
+To apply the current configuration as a predefined configuration, copy this file to the application source directory.
+When the build system sees a file named :file:`prj.partitions.yml` in an applications source directory, it is automatically provided to the partition manager script as the predefined configuration.
+If desired, the file can be modified as described below.
+
+.. _ug_pm_predefined_remove:
+
+Removing a predefined partition
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+A predefined partition is removed by deleting its entry in :file:`prj.partitions.yml`.
+Only partitions adjacent to the ``app`` partition or other removed partitions can itself be removed.
+
+.. _ug_pm_predefined_modify:
+
+Modifying a predefined partition
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+To modify a predefined partition, remove the partition from :file:`prj.partitions.yml` as described above.
+This will add its reserved area to the dynamic partition.
+Now the partition will be configured by Partition Manager.
+If the partition being modified is an image partition, ensure that the build strategy for corresponding sub-image is "Build from source" and update the configured partition size.
+Only partitions adjacent with the ``app`` partition can be modified.
+
+.. _ug_pm_predefined_add_dynamic:
+
+Adding a partition to dynamic partition
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+New partitions are added to the dynamic partition when they are included in the build (i.e. listed in a :file:`pm.yml` file.), but not listed in the predefined configuration.
+
+.. note::
+   Partitions that are included in :file:`prj.partitions.yml` will be ignored if found in the ``placement: before`` or ``placement: after`` property for new partition definitions.
```

Comment:
I'm worried about silently ignoring user-input.

Is there a use-case for having partitions in the prj.partitions.yml that are ignored?

Could we error-out instead?

Reply:
This is the same behavior we have in the normal pm.yml, which is really the same behavior. Note that this is not exactly "user input" as it is typically the author of the sub-image which describe its pm.yml.

I would argue that we have that use case, because one want to use the generic pm.yml provided by the author of the sub-image. This will have several partitions listed to be compatible with different configurations (imagine mcuboot tackling both with and without SPM partition).

Reply:
I thought one would typically copy partitions.yml, not pm.yml, into prj.partitions.yml.

Then the user will anyway have a custom version.

Reply:
But nevertheless, I can see that one would want Kconfig+pm.yml to have precedence over prj.partitions.yml.

Reply:
A user can combine both prj.partitions.yml and pm.yml.